src/Chapter2.hs

```diff
@@ -349,7 +351,10 @@ Implement a function that returns only the first half of a given list.
 "b"
 -}
 -- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
+firstHalf :: [a] -> [a]
+firstHalf l = go l 0 where
+  go (_:_:r) n = go r (succ n)
+  go _       n = take n l
```

Comment:
Interesting implementation :brain: 
Though, you implement an even smarter solution by traversing the same list twice: with step 1 and 2 at the same time. With this approach, you will be able to return list elements immediately, and your solution will work on infinite lists as well.